syntax = "proto3";

option go_package = "services/proto";

message FollowRecommendationUser {
    string handle = 1;
    string host = 2;
    string display_name = 3;
    string bio = 4;
    string image = 5;
    int64 global_id = 6;
}

message FollowRecommendationRequest {
    /* The Rabble integer ID of the user. */
    int64 user_id = 1;
}

message FollowRecommendationResponse {
    enum ResultType {
        OK = 0;
        ERROR = 1;
        // More types can be added here.
    }

    ResultType result_type = 1;

    /* Should only be set if result_type is not OK. */
    string error = 2;

    /* Should only be set if result_type is OK. */
    repeated FollowRecommendationUser results = 3;
}

message UpdateFollowRecommendationsResponse {
    /* Don't return any error, the caller doesn't really care.
       Just log it in the recommender service. */
}

service FollowRecommendations {
  rpc GetFollowRecommendations(FollowRecommendationRequest) returns (FollowRecommendationResponse);
  rpc UpdateFollowRecommendations(FollowRecommendationRequest) returns (UpdateFollowRecommendationsResponse);
}
